lib/apiservers/engine/proxy/archive.go

```diff
@@ -157,23 +156,29 @@ func (a *ArchiveProxy) ArchiveImportWriter(ctx context.Context, store, deviceID
 			params = params.WithFilterSpec(&encodedFilter)
 		}
 
+		var plErr error
 		_, err = a.client.Storage.ImportArchive(params)
 		if err != nil {
 			switch err := err.(type) {
 			case *storage.ImportArchiveInternalServerError:
-				plErr := InternalServerError(fmt.Sprintf("Server error from archive writer for device %s", deviceID))
-				log.Errorf(plErr.Error())
+				plErr = InternalServerError(fmt.Sprintf("Server error from archive writer for device %s", deviceID))
+				op.Errorf(plErr.Error())
 				pipeReader.CloseWithError(plErr)
 			case *storage.ImportArchiveLocked:
-				plErr := ResourceLockedError(fmt.Sprintf("Resource locked for device %s", deviceID))
-				log.Errorf(plErr.Error())
+				plErr = ResourceLockedError(fmt.Sprintf("Resource locked for device %s", deviceID))
```

Comment:
for this and others - general style is to have lower case start for strings in errors as they are frequently concatenated.